storage/innobase/srv/srv0start.cc

```diff
@@ -1668,6 +1665,10 @@ dberr_t srv_start(bool create_new_db)
 
 			srv_log_file_size = srv_log_file_size_requested;
 
+			mysql_mutex_lock(&log_sys.mutex);
+			recv_sys.recovery_on = false;
+			mysql_mutex_unlock(&log_sys.mutex);
```

Comment:
The comment in `log0recv.h` claims that `recv_sys.recovery_on` is protected by `recv_sys.mutex`.

I don’t think that we need any mutex here. When this log resizing is being executed on startup, the buffer pool should be completely idle (no reads or writes pending). You should omit the mutex acquisition here and amend the comment.

Is this assignment really needed here either? If it is, I would suggest to add the assignment near the following statements earlier in this code path:
```c++
			ut_d(recv_no_log_write = true);
			ut_ad(!os_aio_pending_reads());
```
Also, add a comment why the assignment is needed. Normally the assignment is part of `recv_sys.debug_free()`.